### DISCUSSION TITLE
Setting `pyenv global` but `python` still runs the system version

### DISCUSSION THREAD

**Discussion by Danfs64** [pyenv/pyenv#3032] [Q&A] [answered]
I have installed python 3.12 using pyenv (`pyenv install 3.12.5`), and i've set it as global using `pyenv global 3.12.5`.
However, whenever I run `python`, it uses the system 3.10 version of python, and I can't figure out why.
I can access the 3.12 version running `python3.12`, but I feel like setting 3.12 as global should overwrite the `python` executable version
(Yes, I did open a new terminal tab after installing and setting the version)
If anyone can help me, I will really appreciate

Below is the output of some commands when I run them, as reference:
```console
$ pyenv global
> 3.12.5

$ which python
> /home/danfs/.pyenv/shims/python

$ pyenv which python
> /home/danfs/.pyenv/versions/3.12.5/bin/python

$ python --version
> Python 3.10.12
```

edit: typo

**Comment by native-api** [comment]
Run `hash -r` to reset Bash's cache.

https://github.com/pyenv/pyenv/issues/2732

**Reply by Danfs64** [reply]
Thanks for the help. Just found out the problem was not actually with pyenv (but the `hash`command did help me figure out what was). I'll submit an answer clarifying the situation and mark that as an answer

**Comment by Danfs64** [comment, marked as answer]
The problem actually was that I had an alias on my system, pointing `python` to `python3.10`. Only God knows how long that alias was there, for I don't remember making it.

So, if anyone come across a similar problem, remember to check your aliases

Edit: typo
